README.md
![](https://github.com/foex-open-source/apex-builder-extension-by-fos/workflows/build/badge.svg)
<a href="https://chrome.google.com/webstore/detail/apex-builder-extension-by/jhmmfmhnhnnfnejfphieclbibmoaapid">![Chrome Web Store](https://img.shields.io/chrome-web-store/v/jhmmfmhnhnnfnejfphieclbibmoaapid?color=green&label=chrome%20extension)</a>
<a href="https://addons.mozilla.org/en-US/firefox/addon/apex-builder-extension-by-fos/">![Mozilla Add-on](https://img.shields.io/amo/v/apex-builder-extension-by-fos?color=orange&label=firefox%20add-on)</a>

<h1 align="center">APEX Builder Extension by FOS</h1>

<p align="center"><img width="600" src="https://raw.githubusercontent.com/foex-open-source/apex-builder-extension-by-fos/master/misc/image-dark.png"/></p

The APEX Builder Extension by FOS is a free and open-source Chrome & Firefox browser extension that enables the editing of static application, workspace, plug-in and theme files.

## Features

- Editing of static application, workspace, plug-in and theme files directly in the builder
- Leverages the same powerful editor as VS Code
- Allows the editing of multiple files at once. Simply open a new file, and drag the tab
- Save changes without a page submit
- Create new JavaScript, CSS, Less, JSON and HTML files right in the builder
- Minify JavaScript and CSS files, and compile Less files
- Hot Reloading the Runtime Tab when included JS or CSS files are updated

- Syntax highlighting on the Embedded Code page (APEX 20.2)

## Build

Feel free to fork this repo if you'd like to contribute or build your own version.

```bash
# develop
npm install
npm run build         # will create dist/unpacked

# create final extensions
npm install
npm run pack-chrome   # will create dist/chrome.zip
npm run pack-firefox  # will create dist/firefox.zip
```

## Compatibility

- APEX 5.1+
- Last 2 versions of Firefox
- Last 2 versions of Chrome

## Changelog

See [here](https://github.com/foex-open-source/apex-builder-extension-by-fos/blob/master/CHANGELOG.md)

## License

MIT


CHANGELOG.md
## Changelog


### 22.1.0

- APEX 22.1 compatibility
- Page Designer - Modified the Autofill “Items To Submit” field feature to be optional
  - A toggle menu entry is added to the "Settings" menu to enable/disable the feature.
- Page Designer - Comment Out/Uncomment menu entries in the context menu of various components to set the Build Option attribute

### 21.2.0

- APEX 21.2 compatibility
- Full screen option (almost) everywhere
- Page Designer - Quick Save
- Page Designer – Autofill “Items To Submit” field
- Page Designer – Auto dismiss success notification

### 21.1.0

- APEX 21.1 compatibility

### 20.2.0

- APEX 19.2+ Hot Reload for JS & CSS
- Fixed monaco conflict on page 4410 on APEX 20.2 (fallback to old syntax highlighting)

### 20.1.2

- APEX 20.2 specific: Extra Code Editor Options: Enable/Disable Suggestions & Hide/Show Whitespace
- APEX 20.2 specific: Added syntax highlighting on the Embedded Code page

### 20.1.1

- Improved syntax highlighting

### 20.1.0

- Added editing of theme files on page 267
- Syntax highlighting is now based on TextMate grammars
- Save button now saves, compiles and minifies
- Added settings menu. First option: Theme selection
- Upgraded Monaco Editor & Terser
- Added fix to avoid accidental browser "Back" action when horizontally scrolling
- Changed versioning system to match APEX

#### Major changes

- The Firefox add-on does not work offline anymore. Due to a file size restriction of 4MB, Monaco cannot be bundled with the extension anymore (tsWorker.js is the culprit). The library will be loaded through a CDN instead.

### 0.3.0

- added CSS file minification
- added Sourcemap file generation for JS & CSS
- added Plug-in boilerplace code section

### 0.2.1

- Fixed color mode detection on cloud instances


src/editor/data.js
const markup = {};

markup.documentationRegion =
`<div class="a-Region a-Region--noPadding a-Region--sideRegion" id="fos-documentation">
    <div class="a-Region-header">
        <div class="a-Region-headerItems  a-Region-headerItems--title">
            <h2 class="a-Region-title">API Reference</h2>
        </div>
    </div>
    <div class="a-Region-body">
        <div class="a-Region-bodyHeader"></div>
        <ul class="a-LinksList a-LinksList--showArrow">
            <li class="a-LinksList-item">
                <a href="" target="_blank" class="a-LinksList-link fos-plsql-api-link">
                    <span class="a-LinksList-label">PL/SQL</span>
                </a>
            </li>
            <li class="a-LinksList-item">
                <a href="" target="_blank" class="a-LinksList-link fos-js-api-link">
                    <span class="a-LinksList-label">JavaScript</span>
                </a>
            </li>
        </ul>
    </div>
</div>
`;

markup.boilerplateRegion = 
`<div class="a-Region a-Region--noPadding a-Region--sideRegion" id="fos-boilerplate">
    <div class="a-Region-header">
        <div class="a-Region-headerItems a-Region-headerItems--title">
            <h2 class="a-Region-title">Extras</h2>
        </div>
    </div>
    <div class="a-Region-body">
        <div class="a-Region-bodyHeader"></div>
        <ul class="a-LinksList a-LinksList--showArrow">
            <li class="a-LinksList-item">
                <a href="javascript:void();" class="a-LinksList-link fos-boilerplate-link">
                    <span class="a-LinksList-label">Boilerplate Code</span>
                </a>
            </li>
        </ul>
    </div>
</div>
`;

const apiLinks = {
    "50": {
        js: 'https://docs.oracle.com/cd/E59726_01/doc.50/e39149/javascript_api.htm',
        plsql: 'https://docs.oracle.com/cd/E59726_01/doc.50/e39149/toc.htm'
    },
    "51": {
        js: 'https://docs.oracle.com/database/apex-5.1/AEAPI/JavaScript-APIs.htm',
        plsql: 'https://docs.oracle.com/database/apex-5.1/AEAPI/toc.htm',
    },
    "181": {
        js: 'https://docs.oracle.com/database/apex-18.1/AEXJS/index.html',
        plsql: 'https://docs.oracle.com/database/apex-18.1/AEAPI/toc.htm',
    },
    "182": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/18.2/aexjs/index.html',
        plsql: 'https://docs.oracle.com/database/apex-18.2/AEAPI/toc.htm',
    },
    "191": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/19.1/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/19.1/aeapi/index.html',
    },
    "192": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/19.2/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/19.2/aeapi/index.html',
    },
    "201": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/20.1/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/20.1/aeapi/index.html',
    },
    "202": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/20.2/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/20.2/aeapi/index.html',
    },
    "211": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/21.1/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/21.1/aeapi/index.html',
    },
    "212": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/21.2/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/21.2/aeapi/index.html',
    },
    "221": {
        js: 'https://docs.oracle.com/en/database/oracle/application-express/22.1/aexjs/index.html',
        plsql: 'https://docs.oracle.com/en/database/oracle/application-express/22.1/aeapi/index.html',
    }
};

export {markup, apiLinks};